pkg/util/workloads_utils.go

```diff
@@ -137,13 +140,38 @@ func EqualIgnoreHash(template1, template2 *v1.PodTemplateSpec) bool {
 	return apiequality.Semantic.DeepEqual(t1Copy, t2Copy)
 }
 
-func PatchFinalizer(c client.Client, object client.Object, finalizers []string) error {
-	patchByte, _ := json.Marshal(map[string]interface{}{
-		"metadata": map[string]interface{}{
-			"finalizers": finalizers,
-		},
+func UpdateFinalizer(c client.Client, object client.Object, op string, finalizer string) error {
+	switch op {
+	case "Add", "Remove":
+	default:
+		panic(fmt.Sprintf("UpdateFinalizer Func 'op' parameter must be 'Add' or 'Remove'"))
+	}
+
+	key := client.ObjectKeyFromObject(object)
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		fetchedObject := &unstructured.Unstructured{}
+		fetchedObject.SetGroupVersionKind(object.GetObjectKind().GroupVersionKind())
```

Comment:
Don't do this.

1. `object.GetObjectKind().GroupVersionKind()` **might** be empty if the object is from an event or get from client-go or list without deepcopy.
2. Avoid using unstructure object if possible for it will read directly from apiserver.

We can just simplify it with `fetchedObject := object.DeepCopyObject().(client.Object)`, and then Get() will overwrite the content of the object.